source/adaptive_load/metrics_evaluator_impl.cc

```diff
@@ -15,16 +15,42 @@ using ::nighthawk::adaptive_load::MetricSpec;
 using ::nighthawk::adaptive_load::MetricSpecWithThreshold;
 using ::nighthawk::adaptive_load::ThresholdSpec;
 
+// Extract estimated measuring period from Nighthawk output proto.
+absl::StatusOr<MeasuringPeriod> GetMeasuringPeriod(const nighthawk::client::Output& output) {
+  if (output.results().empty()) {
+    return absl::Status{absl::StatusCode::kInvalidArgument, "output.results cannot be empty."};
+  }
+  // Assume the first result is a good representation for the entire run.
```

Comment:
I remember we discussed sorting these and choosing one in particular.

Why is it safe to assume the first is a good representation? Since this is concurrent, I don't think there are any guarantees.

Reply:
Changed to a more accurate approach. I originally opted for using the first one as the deviation seemed to be not large and any result seemed like a good enough representation.

Reply:
I think the current approach will work well. Thank you.